src/diff/children.js

```diff
@@ -75,12 +75,16 @@ export function diffChildren(parentDom, newParentVNode, oldParentVNode, context,
 						oldChildren[j] = undefined;
 						break;
 					}
-					oldVNode = null;
+					oldVNode = EMPTY_OBJ;
 				}
 			}
 
 			// Morph the old element into the new one, but don't append it to the dom yet
-			newDom = diff(parentDom, childVNode, oldVNode, context, isSvg, excessDomChildren, mounts, ancestorComponent, null, oldDom);
+			newDom = diff(parentDom, childVNode, oldVNode || EMPTY_OBJ, context, isSvg, excessDomChildren, mounts, ancestorComponent, null, oldDom);
+
+			if ((j = childVNode.ref) && (oldVNode || EMPTY_OBJ).ref != j) {
```

Comment:
Do you think we could shave more bytes by always defaulting the `oldVNode` to `EMPTY_OBJ`? Therefore we would not need to have the `OR` in here?

Reply:
Saved 2 bytes! In order for null placeholders to work, if `oldVNode` is exactly `null`, it needs to stay that way while trying to find a match. So I coerce it to `EMPTY_OBJ` after the matching is finished.